src/main/java/net/snowflake/ingest/streaming/internal/LiteralQuoteUtils.java

```diff
@@ -0,0 +1,170 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+import net.snowflake.ingest.utils.Utils;
+
+/**
+ * Util class to normalise literals to match server side metadata.
+ *
+ * <p>Note: The methods in this class have to be kept in sync with the respective methods on server
+ * side.
+ */
+class LiteralQuoteUtils {
+  private static final String[] reservedKeywords = {
```

Comment:
Question: should we even maintain this list? Let's say that we don't do any checks w/ respect to these. The operation should fail server side at registration time, right?

Reply:
If a user creates table with column "CREATE", for example, which is a valid name, and then ingests a row with `row.put("CREATE", 1)`, we need to quote the input to match it with the column display name.